### PR TITLE
official java solution for grass planting

### PR DESCRIPTION
**Pull request by nwatx**

### CODE REVIEW

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
Tabs

File: solutions/USACO jan19-grass-planting.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+id: usaco-849
+source: USACO Silver 2019 January
+title: Grass Planting
+author: Neo Wang
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_planting_silver_jan19.html)
+
```
Comment (HarryWangATX) [inline solutions/USACO jan19-grass-planting.mdx:9]: time complexity

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
https://usaco.guide/general/contributing?lang=cpp#adding-a-solution
step 4

File: solutions/USACO jan19-grass-planting.mdx (hunk 1 above)
Comment (HarryWangATX) [reply solutions/USACO jan19-grass-planting.mdx:9]: wait oops mb

File: solutions/USACO jan19-grass-planting.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+<LanguageSection>
+
+<JavaSection>
+
+## Solution 1 (Official)
```
Comment (HarryWangATX) [inline solutions/USACO jan19-grass-planting.mdx:14]: just change to java impl

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

**Review by HarryWangATX — APPROVED** [review]
doesnt this conflict with #797 lol?

**Comment by dongliu0426** [comment]
i think this solutions is much more cleaner imo

**Comment by HarryWangATX** [comment]
> i think this solution is much more cleaner imo

👍

File: solutions/USACO jan19-grass-planting.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+[Official Analysis](http://www.usaco.org/current/data/sol_planting_silver_jan19.html)
+
+**Time Complexity: $\mathcal{O}(N)$**
+
+<LanguageSection>
+
+<JavaSection>
+
+## Solution 1 (Java)
```
Comment (frodakcin) [inline solutions/USACO jan19-grass-planting.mdx:16]: I think time complexity should be directly beneath the solution header.

In other words, either move "Solution 1" up or move TC down. Also the thing in $$'s (I think) shouldn't be bolded. Just write `**Time Complexity:** $\mathcal{O}(N)$`

**Review by frodakcin — CHANGES_REQUESTED** [review]
otherwise ok

**Review by Farrius — APPROVED** [review]

**Review by frodakcin — APPROVED** [review]

**Pull request merged**